drgn_tools/workqueue.py

```diff
@@ -63,6 +94,33 @@ def _print_work(work: Object) -> None:
     )
 
 
+def workqueue_get_pwq(workqueue: Object, cpu: int) -> Object:
+    """
+    Find pool_workqueue of a bound workqueue for a given CPU.
+
+    :param workqueue: ``struct workqueue_struct *``
+    :return: ``struct pool_workqueue *``.
+    """
+    # At first Linux kernel commit ee1ceef72754 ("workqueue: Rename wq->cpu_pwqs to
+    # wq->cpu_pwq") (in v6.6) renamed cpu_pwqs to cpu_pwq and then Linux kernel commit
+    # 687a9aa56f81("workqueue: Make per-cpu pool_workqueues allocated and released
+    # like unbound ones") (in v6.6) changed cpu_pwq to double pointer.
+    # As both of the changes were made in v6.6, there are no kernel versions
+    # with wq->cpu_pwq as a pointer. Still I have mentioned both the changes so that
+    # we can track both name change and type change of this member.
+    try:
+        pwq = per_cpu_ptr(workqueue.cpu_pwqs, cpu)
+    except AttributeError:
+        prog = workqueue.prog_
+        pwq = Object(
+            prog,
+            "struct pool_workqueue",
+            address=prog.read_word(per_cpu_ptr(workqueue.cpu_pwq, cpu)),
+        ).address_of_()
```

Comment:
You can simplify this by just dereferencing the double pointer.
```suggestion
        pwq = per_cpu_ptr(workqueue.cpu_pwq, cpu)[0]
```

Reply:
Makes sense. Have done that now.